rs/bitcoin/validation/src/header.rs

```diff
@@ -48,26 +42,13 @@ pub fn validate_header(
     store: &impl HeaderStore,
     header: &BlockHeader,
 ) -> Result<(), ValidateHeaderError> {
-    let chain_height = store.get_height();
     let (prev_header, prev_height) = match store.get_header(&header.prev_blockhash) {
         Some(result) => result,
         None => {
             return Err(ValidateHeaderError::PrevHeaderNotFound);
         }
     };
 
-    if !is_header_within_one_year_of_tip(prev_height, chain_height) {
-        return Err(ValidateHeaderError::HeightTooLow);
-    }
-
-    if !is_timestamp_valid(store, header) {
-        return Err(ValidateHeaderError::HeaderIsOld);
-    }
-
-    if !is_checkpoint_valid(network, prev_height, header, chain_height) {
```

Comment:
Can you elaborate a bit more why it is safe to remove this check ?

How hard it is for a bad BTC peer to fake a BTC chain that the adapter will accept ?

Reply:
A BTC peer would need to do all the work of creating a chain of "valid" blocks, which includes the PoW. 

Even if we keep the checkpoints, faking a chain starting from the most recent checkpoint would be at least as easy as faking a chain from an older point.